hooks/useGetUserLocation.ts

```diff
@@ -0,0 +1,21 @@
+import { useState } from "react";
+
+export function useGetUserLocation() {
+  const [location, setLocation] = useState<
+    { latitude: number; longitude: number } | undefined
+  >(undefined);
+
+  const successCallback = (position: any) => {
+    setLocation(position.coords);
+  };
+
+  const errorCallback = () => {
+    alert("Konum belirleme için lokasyon bilgisine izin vermelisiniz.");
+  };
+
+  const getUserLocation = () => {
+    navigator.geolocation.getCurrentPosition(successCallback, errorCallback);
```

Comment:
Her ihtimale karsi, navigator'u kontrol etmekte fayda var:

```suggestion
if (navigator)
    navigator.geolocation.getCurrentPosition(successCallback, errorCallback);
```